libs/gretty/build.gradle

```diff
@@ -1,4 +1,5 @@
 plugins {
+  id 'java-gradle-plugin'
```

Comment:
I can't really see the implications of the plugin - they might be manifold. Is the desired effect the same as 80d8dddc5a0fa5744e77755044043faad0bf316c? Or setting `org.gradle.warning.mode=fail` for that matter?

Reply:
I removed the plugin application. I'll do that in a separate PR.

In short, this plugin sets up the build for building a Gradle plugin. So it configures publishing tasks and test tasks and the like. More importantly for this PR, it creates a `validatePlugins` task which checks that all tasks are annotated correctly.

Reply:
> So it configures publishing tasks and test tasks and the like.

Yes, I suspected that. In theory we have all of that, the patch replace / beef up the existing packaging needs to take that into account. I think it needs a little more coordination. Also see #192, @javabrett wants to touch publishing anyway.